### PR TITLE
Update/upgrade plans

### PR DESCRIPTION
**Pull request by alexeygorobets**
### What changes were proposed in this pull request?
Resolves [D2IQ-74329](https://jira.d2iq.com/browse/D2IQ-74329)
Update and upgrade plans implementation:
- added upgrade plan with cleanup steps for webhook and spark deployment
- update plan covered by deploy plan implementation
- KUTTL tests: upgrade from the current catalog version to current version in the branch, update log level 

### Why are the changes needed?
to be able to upgrade and  update KUDO Spark

### How were the changes tested?
on the local cluster

### CODE REVIEW

File: kudo-spark-operator/operator/operator.yaml
Comment (alembiewski) [inline kudo-spark-operator/operator/operator.yaml:13]: should we add `rc1` or `dev` suffix here?

File: kudo-spark-operator/operator/operator.yaml
Comment (alembiewski) [inline kudo-spark-operator/operator/operator.yaml:121]: not sure if it's really necessary to delete operator deployment as a part of the upgrade - I believe KUDO should be able to update the existing deployment and then k8s will do the rest

File: tests/test-upgrade-and-update/02-update.yaml
Comment (alembiewski) [inline tests/test-upgrade-and-update/02-update.yaml:4]: do you think it makes sense to extend this test-case and add more parameters just to exercise and cover more of the operator's functionality? Thinking about enabling Spark History server or metrics in the running instance of the operator and the related resources are applied or updated accordingly.

**Review by alembiewski — COMMENTED** [review]
Thanks, @alexeygorobets! Overall LGTM, left a few comments.

File: kudo-spark-operator/operator/operator.yaml
Comment (alexeygorobets) [reply kudo-spark-operator/operator/operator.yaml]: thanks, removed that

File: tests/test-upgrade-and-update/02-update.yaml
Comment (alexeygorobets) [reply tests/test-upgrade-and-update/02-update.yaml]: replaced this with enabling history server

**Review by akirillov — APPROVED** [review]
thanks, @alexeygorobets. this LGTM 👍  I'd suggest waiting for @alembiewski feedback before merging.

File: tests/test-upgrade-and-update/02-update.yaml
Comment (alembiewski) [inline tests/test-upgrade-and-update/02-update.yaml:10]: Can we use a local folder here? The current approach seems to give false-negative results given the [following error in CI](https://teamcity.mesosphere.io/buildConfiguration/Frameworks_DataServices_Kudo_Spark_Pr_SparkPrKonvoyKudoV0100/3084769?buildTab=log&focusLine=16417&linesState=16417):
```
logger.go:42: 12:39:14 | test-upgrade-and-update/2-update | cat: .tmp/s3_bucket_path: No such file or directory

```

**Review by alembiewski — APPROVED** [review]
Looks great, @alexeygorobets! Left one comment about using the local folder for `historyServerFsLogDirectory`.

File: tests/test-upgrade-and-update/02-update.yaml
Comment (alexeygorobets) [reply tests/test-upgrade-and-update/02-update.yaml]: changed to the local folder

**Pull request merged**
